pkg/kubectl/cmd/delete/delete.go

```diff
@@ -216,6 +219,34 @@ func (o *DeleteOptions) Validate() error {
 	case o.ForceDeletion:
 		fmt.Fprintf(o.ErrOut, "warning: --force is ignored because --grace-period is not 0.\n")
 	}
+
+	if o.WarnClusterScope {
```

Comment:
move this inside the Visit method inside DeleteResult so we don't iterate through all objects twice

Reply:
If I moved them inside Visit, there is warning message for each cluster-scoped resource.

It turns out there are multiple warning message in the output.

Reply:
Set a warnedClusterScope boolean outside the visit function and set it to true once you've warned once